components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/OrganizationManagementDAOImpl.java

```diff
@@ -709,97 +728,101 @@ public String resolveTenantDomain(String organizationId) throws OrganizationMana
+                        } else {
+                            Organization organization = buildOrganization(resultSet);
+                            organizationMap.put(organization.getId(), organization);
+                            return organization;
+                        }
+                    },
+                    namedPreparedStatement -> setPreparedStatementParams(namedPreparedStatement, organizationId,
+                            applicationAudience, limit, filterQueryBuilder, parentIdFilterQueryBuilder, permissions,
+                            permissionPlaceholder, userID));
+            organizations = new ArrayList<>(organizationMap.values());
+        } catch (DataAccessException e) {
+            throw handleServerException(ERROR_CODE_ERROR_RETRIEVING_ORGANIZATIONS, e);
         }
+        return organizations;
+    }
 
+    private Organization buildOrganization(ResultSet resultSet) throws SQLException {
+
+        Organization organization = new Organization();
+        organization.setId(resultSet.getString(1));
+        organization.setName(resultSet.getString(2));
+        organization.setCreated(resultSet.getTimestamp(3).toInstant());
+        organization.setStatus(resultSet.getString(4));
+        return organization;
+    }
+
+    private List<BasicOrganization> getOrganizationsBasicInfo(boolean authorizedSubOrgsOnly, boolean recursive,
+                                                              Integer limit, String organizationId, String sortOrder,
+                                                              List<ExpressionNode> expressionNodes,
+                                                              List<ExpressionNode> parentIdExpressionNodes,
+                                                              String applicationAudience)
+            throws OrganizationManagementServerException {
+
+        FilterQueryBuilder filterQueryBuilder = buildFilterQuery(expressionNodes);
+        FilterQueryBuilder parentIdFilterQueryBuilder = buildParentIdFilterQuery(parentIdExpressionNodes);
+
+        String userID =  getUserId();
+        String sqlStmt = prepareGetOrganizationQuery(authorizedSubOrgsOnly, recursive, sortOrder,
+                applicationAudience, filterQueryBuilder, parentIdFilterQueryBuilder, userID);
 
-        if (StringUtils.isBlank(parentIdFilterQuery)) {
-            sqlStmt += filterQueryBuilder.getFilterQuery() +
-                    String.format(getOrgSqlStmtTail, SET_ID, recursive ? "> 0" : "= 1", sortOrder);
-        } else {
-            sqlStmt += filterQueryBuilder.getFilterQuery() +
-                    String.format(getOrgSqlStmtTail, parentIdFilterQuery, recursive ? "> 0" : "= 1",
-                            sortOrder);
-        }
         List<String> permissions;
         String permissionPlaceholder;
-        if (CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME) {
+        if (Boolean.TRUE.equals(CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME)) {
```

Comment:
```suggestion
        if (CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME) {
```
Can't we go with this?

Reply:
This was changed to avoid using boxed Boolean types directly in boolean expressions, as they may throw a NullPointerException if the value is null. However, this is not an issue if it has already been null-checked.